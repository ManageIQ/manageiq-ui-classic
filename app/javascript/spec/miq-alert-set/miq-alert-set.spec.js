import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import MiqStructuredList from '../../components/miq-structured-list';
import {
  infoData, alertSetAssignedData, alertSetNotes, alertSetAlerts,
} from './miq-alert-set.data';

describe('MiqAlertSetStructuredList', () => {
  it('should render alert set info structured list', () => {
    const wrapper = shallow(<MiqStructuredList
      rows={infoData.items}
      title={infoData.title}
      mode={infoData.mode}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render alert set assigned structured list', () => {
    const wrapper = shallow(<MiqStructuredList
      rows={alertSetAssignedData.items}
      title={alertSetAssignedData.title}
      mode={alertSetAssignedData.mode}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render alert set notes structured list', () => {
    const wrapper = shallow(<MiqStructuredList
      rows={alertSetNotes.items}
      title={alertSetNotes.title}
      mode={alertSetNotes.mode}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render alert set alerts structured list', () => {
    const wrapper = shallow(<MiqStructuredList
      rows={alertSetAlerts.items}
      title={alertSetAlerts.title}
      mode={alertSetAlerts.mode}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
