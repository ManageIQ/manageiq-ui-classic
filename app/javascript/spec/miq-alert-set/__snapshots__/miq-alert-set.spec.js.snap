// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiqAlertSetStructuredList should render alert set alerts structured list 1`] = `
<Accordion
  align="start"
  className="miq-structured-list-accordion miq_alert_set"
>
  <AccordionItem
    open={true}
    title="Alerts"
  >
    <FeatureToggle(StructuredListWrapper)
      ariaLabel="Structured list"
      className="miq-structured-list miq_alert_set"
    >
      <MiqStructuredListBody
        clickEvents={false}
        mode="miq_alert_set"
        rows={
          Array [
            Object {
              "cells": Array [
                Object {
                  "icon": "pficon pficon-warning-triangle-o",
                  "value": "Alert description",
                },
              ],
              "onclick": Object {
                "url": "/miq_alert/show/1",
              },
              "title": "View this Alert",
            },
          ]
        }
      />
    </FeatureToggle(StructuredListWrapper)>
  </AccordionItem>
</Accordion>
`;

exports[`MiqAlertSetStructuredList should render alert set assigned structured list 1`] = `
<Accordion
  align="start"
  className="miq-structured-list-accordion miq_alert_set_assigned"
>
  <AccordionItem
    open={true}
    title="Assigned To"
  >
    <FeatureToggle(StructuredListWrapper)
      ariaLabel="Structured list"
      className="miq-structured-list miq_alert_set_assigned"
    >
      <MiqStructuredListBody
        clickEvents={false}
        mode="miq_alert_set_assigned"
        rows={
          Array [
            Object {
              "label": "The Enterprise",
            },
            Object {
              "label": "Object class",
              "value": Array [
                Object {
                  "value": "path1",
                },
                Object {
                  "value": "path2",
                },
                Object {
                  "value": "path3",
                },
              ],
            },
          ]
        }
      />
    </FeatureToggle(StructuredListWrapper)>
  </AccordionItem>
</Accordion>
`;

exports[`MiqAlertSetStructuredList should render alert set info structured list 1`] = `
<Accordion
  align="start"
  className="miq-structured-list-accordion miq_alert_set_info"
>
  <AccordionItem
    open={true}
    title="Basic Information"
  >
    <FeatureToggle(StructuredListWrapper)
      ariaLabel="Structured list"
      className="miq-structured-list miq_alert_set_info"
    >
      <MiqStructuredListBody
        clickEvents={false}
        mode="miq_alert_set_info"
        rows={
          Array [
            Object {
              "label": "Description",
              "value": "bar",
            },
            Object {
              "label": "Mode",
              "value": "Mode",
            },
          ]
        }
      />
    </FeatureToggle(StructuredListWrapper)>
  </AccordionItem>
</Accordion>
`;

exports[`MiqAlertSetStructuredList should render alert set notes structured list 1`] = `
<Accordion
  align="start"
  className="miq-structured-list-accordion miq_alert_set_notes"
>
  <AccordionItem
    open={true}
    title="Notes"
  >
    <FeatureToggle(StructuredListWrapper)
      ariaLabel="Structured list"
      className="miq-structured-list miq_alert_set_notes"
    >
      <MiqStructuredListBody
        clickEvents={false}
        mode="miq_alert_set_notes"
        rows={
          Array [
            Object {
              "value": Object {
                "input": "text_area",
                "readonly": true,
                "rows": 4,
                "text": "text area description",
              },
            },
          ]
        }
      />
    </FeatureToggle(StructuredListWrapper)>
  </AccordionItem>
</Accordion>
`;
